### PR TITLE
Bookstore: add reducers and actions

### PR DESCRIPTION
**Pull request by cilfonegabriel**
# Dear Code Reviewers Team,
### In this part of the project:

- I have configured my Redux Store and written actions and reducer for adding and removing books.

- Added Redux Toolkit(npm install react-redux @reduxjs/toolkit).

- Defined action types for adding and removing a book.

- Set the initial state to be an empty array of books.

- Export Action Creators for your actions.

- Write your reducer and export it as default.

- Define state changes for the actions that you created.

- In case of unknown action - return the current state.

- Define an action type for checking the status.

- Set the initial state to be an empty array of categories.

- Export Action Creators for your actions.

- Write your reducer and export it as default.

- For the check status action return a string "Under construction".

- In case of unknown action - return to the current state.

- importing the necessary methods from Redux Toolkit.

- Combine both reducers into a root reducer by using configureStore function.

### CODE REVIEW

File: src/redux/book/book.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+const ADD_A_BOOK = 'bookStore/book/ADD_A_BOOK';
+const REMOVE_A_BOOK = 'bookStore/book/REMOVE_A_BOOK';
+const LIST_BOOKS = 'bookStore/book/LIST_BOOKS';
+
+const initialState = [{
+  id: '1', title: 'bealy', author: 'father', category: 'java',
+}];
```
Comment (gemmen29) [inline src/redux/book/book.js]: - [ ] Kindly set the `intialState` with an empty array of books as per this [requirement](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/add_reducers_an_actions.md#:~:text=Set%20the%20initial%20state%20to%20be%20an%20empty%20array%20of%20books.).

File: src/redux/book/book.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
+const ADD_A_BOOK = 'bookStore/book/ADD_A_BOOK';
+const REMOVE_A_BOOK = 'bookStore/book/REMOVE_A_BOOK';
+const LIST_BOOKS = 'bookStore/book/LIST_BOOKS';
```
Comment (gemmen29) [inline src/redux/book/book.js:3]: - [ ] [OPTIONAL] Kindly remove the `LIST_BOOKS` action creator as it's redundant. It's the same as the default case.

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi @cilfonegabriel, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by julie-ify — APPROVED** [review]
Hi @cilfonegabriel,

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- [ ] **[OPTIONAL]** I would recommend you fix your project title on your readme file


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**